### ISSUE TITLE
OAuth2 authentication failure

### ISSUE THREAD

**Issue by pavanelthepumisc** [gravitee-io/issues#4687]
I've setup an API with a plan of OAuth2 authentication and subscribed to it. When I'm trying to access the API from post man I'm getting following error:

![image](https://user-images.githubusercontent.com/29723106/100580767-8e99cb80-330c-11eb-8b7e-52b37720e74a.png)

### Expected Behavior
As my token is valid, I should get response from my API

### Current Behavior
Getting error

### Steps to Reproduce (for bugs)

Created API
Added a Resource with Generic OAuth2 Authorization Server type
Added a plan with the OAuth Resource created in step 2
Subscribed to it
Got my access token by hitting my keycloak server from postman
Accessing API with the access token received in step 5

### Your Environment
* Version used: 3.4.0
* Browser Name and version: Chrome 79.0.3945.130 (Official Build) (64-bit)
* Operating System and version: MacOS Catalina Version 10.15.6

**Comment by aelamrani** [comment]
Hi @pavanelthepumisc 

Which version of Keycloak are you using?

**Comment by phenompavan** [comment]
11.0.2

**Comment by pavanelthepumisc** [comment]
11.0.2

**Comment by aelamrani** [comment]
Any error in gateway logs?

**Comment by pavanelthepumisc** [comment]
No errors in logs @aelamrani

**Comment by aelamrani** [comment]
So, just had a try and everything is working on my side. Can you share the configuration of the Gravitee resource?

**Comment by pavanelthepumisc** [comment]
Here is my configuration of the Gravitee resource @aelamrani 

![image](https://user-images.githubusercontent.com/29723106/100967821-32c28300-3556-11eb-8c45-302ccd3c1ef2.png)
![image](https://user-images.githubusercontent.com/29723106/100967853-440b8f80-3556-11eb-99d3-43ae159737fd.png)
![image](https://user-images.githubusercontent.com/29723106/100967903-58e82300-3556-11eb-9a1c-61690097fb32.png)

**Comment by aelamrani** [comment]
Ok try something like this one instead: 

<img width="254" alt="Capture d’écran 2020-12-03 à 06 48 35" src="https://user-images.githubusercontent.com/1580262/100969626-a69f6400-3533-11eb-97b8-3e8c4f19de9b.png">

**Comment by pavanelthepumisc** [comment]
Here is my configuration @aelamrani 

![image](https://user-images.githubusercontent.com/29723106/100973363-6c988700-3560-11eb-8866-6c91f770e069.png)

**Comment by aelamrani** [comment]
Cool and works now?

**Comment by aelamrani** [comment]
Here is the full auth server URL in my example: http://localhost:8080/auth/realms/master/protocol/openid-connect

**Comment by pavanelthepumisc** [comment]
No, it's not working @aelamrani . Getting following error:

![image](https://user-images.githubusercontent.com/29723106/100975081-753e8c80-3563-11eb-9086-7e864ada80c2.png)

**Comment by pavanelthepumisc** [comment]
Here is my API export @aelamrani

**Comment by aelamrani** [comment]
Ok, can you check that your introspect call return well the `client_id` defined on the Gravitee.io application used to subscribe?

**Comment by pavanelthepumisc** [comment]
Yes @aelamrani 

Here is the response from my introspect URL with the access token, client_id and client_secret:

`{
    "exp": 1606982582,
    "iat": 1606982282,
    "jti": "47246931-68d9-4a51-ae5c-89d57c7f2bc6",
    "iss": "https://auth-dev.custompro.com/auth/realms/APIGateway",
    "aud": "account",
    "sub": "76e8f020-a9d8-48b1-917b-9a1a9a3fe0e8",
    "typ": "Bearer",
    "azp": "HILTGLOBAL",
    "session_state": "3e7b98a8-ca2e-44be-bf0b-029af2fb7335",
    "preferred_username": "service-account-hiltglobal",
    "email_verified": false,
    "acr": "1",
    "allowed-origins": [
        "http://localhost:8001"
    ],
    "realm_access": {
        "roles": [
            "offline_access",
            "uma_authorization"
        ]
    },
    "resource_access": {
        "account": {
            "roles": [
                "manage-account",
                "manage-account-links",
                "view-profile"
            ]
        }
    },
    "scope": "profile email",
    "clientHost": "157.48.213.199",
    "clientId": "HILTGLOBAL",
    "clientAddress": "157.48.213.199",
    "client_id": "HILTGLOBAL",
    "username": "service-account-hiltglobal",
    "active": true
}`

**Comment by aelamrani** [comment]
Are you calling the same endpoint to get the token and for the introspection?

**Comment by pavanelthepumisc** [comment]
Yes @aelamrani

**Comment by fabMrc** [comment]
I have the same issue so I am moving to JWT plan it is a best suited way for me. 
However I am following this issue if you need more information I can help you

**Comment by pavanelthepumisc** [comment]
@aelamrani can you help with this issue, please?

FYI @brasseld

**Comment by brasseld** [comment]
@pavanelthepumisc the error message is about nginx, don't seem to be related to gravitee.

**Comment by pavanelthepumisc** [comment]
But I'm getting this error only with OAuth2 plan @brasseld. The same API is working with API Key plan. Please help.

**Comment by brasseld** [comment]
Could it be a response from your backend ?

**Comment by pavanelthepumisc** [comment]
Nope. I'm expecting Unautorized in the first place as I passed wrong token. Even if I pass correct token, same error. https://dummy.restapiexample.com/api/v1/employees This is my backend @brasseld

**Comment by brasseld** [comment]
So why do we have this nginx response ? Can you describe a bit your architecture ?

**Comment by pavanelthepumisc** [comment]
It's pretty simple @brasseld . For testing OAuth2 plan, I followed the following steps:

1. Created API (with backend server: https://dummy.restapiexample.com/api/v1/employees)
2. Added a plan for this API with OAuth2 security
3. Added Generic OAuth2 Authorization Server
4. Now, trying to access  my API and getting this error

Here is my Generic OAuth2 Authorization Server configuration:

![image](https://user-images.githubusercontent.com/29723106/127160910-d7598a3b-82f2-45dc-8a39-68860f7e1af4.png)
![image](https://user-images.githubusercontent.com/29723106/127160937-c014c2d7-0c5e-4dc9-bd92-97c3fda27956.png)

**Comment by pavanelthepumisc** [comment]
I'm getting this error when I send the Bearer Token. If I don't send any token, I'm getting a 401-Unauthorised error. @brasseld
